damus/Nostr/NostrEvent.swift

```diff
@@ -832,6 +845,10 @@ func first_eref_mention(ev: NostrEvent, privkey: String?) -> Mention? {
     return nil
 }
 
+func isFriend(pubkey: String) -> Bool {
+    return Contacts(our_pubkey: pubkey).is_friend(pubkey)
```

Comment:
Thanks for the feedback! I've been unplugged over the weekend but I'll get on this now.

Reply:
I've refactored the changes but I still can't figure out how to access the DamusState environment variable from the NostrEvent  file without completely reinitializing the class. Do you know how this could be done? The usual

```swift
let state: DamusState
```

returns `Constant 'state' used before being initialized`